gemini/use-cases/transcription-summarization/diarized_transcription_and_summarization.ipynb

```diff
@@ -0,0 +1,931 @@
+        "                    ex_output_transcription=ex_output_transcription\n",
+        "                )\n",
+        "            )\n",
+        "        audio_url = url_audio_chunks[i]\n",
+        "        audio_file = Part.from_uri(audio_url, mime_type=file_format)\n",
+        "        ex_audio_file = Part.from_uri(path_example_audio, mime_type=file_format)\n",
+        "        contents = [prompt, ex_audio_file, prompt_2, audio_file]\n",
+        "        response = generation_model_audio.generate_content(\n",
+        "            contents=contents, generation_config=generation_config_audio\n",
+        "        )\n",
+        "        generated_transcription += response.text\n",
+        "\n",
+        "    return generated_transcription"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "1b4e256e8259"
+      },
+      "source": [
+        "### Step 3 and 4: Fact extraction & Summary generation\n",
+        "Step 3 involves identifying key information related to the specific task/dsocument ( in this example financial advisory document) that needs to be completed. The model is prompted to recognize and extract crucial details such as client names, investment preferences, risk tolerance, and financial goals.\n",
+        "\n",
+        "Step 4 then focuses on generating concise and accurate summaries for each field within the document. Leveraging the extracted facts from the previous step and employing Zero-shot Chain-of-Thought (CoT) prompting, Gemini creates multiple  summaries tailored to the specific domain and the requirements of each form field. This ensures the generated summaries are not only informative but also comply any internal guidelines requirements.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "82811a63412e"
+      },
+      "outputs": [],
+      "source": [
+        "def extract_facts(generated_transcription: str, quick: bool = True):\n",
+        "    \"\"\"\n",
+        "    Take the generated transcript and extract key domain/task-specific facts for downstream processing\n",
+        "    \"\"\"\n",
+        "    generation_model_facts = GenerativeModel(\n",
+        "        \"gemini-1.5-pro-002\".replace(\"pro\", \"flash\" if quick else \"pro\")\n",
+        "    )\n",
+        "    generation_config_facts = GenerationConfig(\n",
+        "        temperature=0.0,\n",
+        "        max_output_tokens=512,\n",
+        "    )\n",
+        "\n",
+        "    prompt = \"\"\"You are an experienced bank advisor who provides consultations to clients regarding financial products. You conducted a telephone consultation with a client, during which you discussed the client's need for one or more financial products. This consultation was recorded. A consultation can span multiple individual conversations. The transcription or transcriptions of the recording can be found below.\n",
```

Comment:
I wonder if it would make sense to move these prompts into dedicated files using the [prompt file format](https://humanloop.com/docs/v5/reference/prompt-file-format)

You could download using the github raw links, and it would keep the notebook simpler/easier to follow.

Reply:
hi holt, good suggestion however i think for this use case we want the user to read it directly in code  to understand how we tackled some of the problems mentioned and to be able to modify it on the fly, all of which is easier if the prompt is inline